### PR TITLE
Batch padding for DDP

### PR DESCRIPTION
**Pull request by psinger**
Pads the quantile across all batches across all GPUs instead of separately per GPU

### CODE REVIEW

File: llm_studio/src/utils/data_utils.py [hunk 1]
```diff
@@ -656,33 +657,24 @@ def batch_padding(
...
+            lengths = torch.stack(
+                [
+                    torch.where(batch[mask_key][i] == 1)[0].max()
+                    for i in range(batch[mask_key].size(0))
+                ]
+            ).float()
+        if cfg.environment._distributed:
+            lengths = sync_across_processes(lengths, cfg.environment._world_size)
+        idx = int(
+            torch.floor(torch.quantile(lengths, 1 - cfg.tokenizer.padding_quantile))
```
[the remark this answers is not in the document]
Comment (psinger) [reply llm_studio/src/utils/data_utils.py]: good catch, thx, fixed
